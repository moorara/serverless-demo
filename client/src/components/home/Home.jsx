import React from 'react'

import Message from './Message'

const Home = (props) => (
  <Message content="Hello, World!" />
)

export default Home
