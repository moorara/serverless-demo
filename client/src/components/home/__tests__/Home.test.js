import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('../../../api')
import Api from '../../../api'
import Home from '../Home'

describe('Home', () => {
  let setIsLoading

  beforeEach((() => {
    Api.reset()
    setIsLoading = jest.fn()
  }))

  test('renders react component when api call fails', done => {
    Api.mock.getMessage.error = 'api failure is mocked!'
    const component = renderer.create(<Home setIsLoading={setIsLoading} />)

    // Wait for componentDidMount
    setImmediate(() => {
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()

      expect(setIsLoading.mock.calls.length).toBe(2)
      expect(setIsLoading.mock.calls[0][0]).toBe(true)
      expect(setIsLoading.mock.calls[1][0]).toBe(false)
      done()
    })
  })
  test('renders react component when api call succeeds', done => {
    Api.mock.getMessage.message = 'Hello, Mock!'
    const component = renderer.create(<Home setIsLoading={setIsLoading} />)

    // Wait for componentDidMount
    setImmediate(() => {
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()

      expect(setIsLoading.mock.calls.length).toBe(2)
      expect(setIsLoading.mock.calls[0][0]).toBe(true)
      expect(setIsLoading.mock.calls[1][0]).toBe(false)
      done()
    })
  })
})
