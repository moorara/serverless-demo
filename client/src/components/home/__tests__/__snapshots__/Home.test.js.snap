// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders react component when api call fails 1`] = `
<section
  className="hero is-info is-medium"
>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        Message Panel
      </h1>
      <div
        className="content"
      >
        <p>
          
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`Home renders react component when api call succeeds 1`] = `
<section
  className="hero is-info is-medium"
>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        Message Panel
      </h1>
      <div
        className="content"
      >
        <p>
          Hello, Mock!
        </p>
      </div>
    </div>
  </div>
</section>
`;
